src/lib.rs

```diff
@@ -2,3 +2,117 @@
 #![cfg_attr(feature = "nightly", feature(external_doc))]
 #![cfg_attr(feature = "nightly", doc(include = "../README.md"))]
 #![cfg_attr(test, deny(warnings))]
+
+extern crate clap;
+
+use clap::{App, AppSettings, Arg, ArgSettings, SubCommand};
+
+fn main() {
+    let mut a = App::new("testapp")
+        .about("Pointless application")
+        .setting(AppSettings::SubcommandRequiredElseHelp)
+        .author("Katharina Fey <kookie@spacekookie.de>")
+        // .author("Yosh Wuyts <y@w.s") 
+        .long_about("Lorem Ipsum bla bla bla")
+        .arg(Arg::with_name("debug").short("d").help("Make program output debug messages"))
+        .arg(Arg::with_name("output").short("o").takes_value(true).help("Output File"))
+        .subcommand(SubCommand::with_name("foo").arg(Arg::with_name("bar").short("b").long("barr")));
+
+    // println!("{:#?}", a);
+    let manual = Manual::from_clap(&a);
+    println!("{:#?}", manual);
+}
+
+/// Describe an argument or option
+#[derive(Debug)]
+pub struct ManualArg {
+    short: Option<char>,
+    long: Option<String>,
```

Comment:
Is there a plan for handling options that are available under multiple names?

What about options that also have a negated form, `--[no]-do-the-thing`?

Reply:
Oh yea, we should definitely scope aliases properly.

In the end this is a little bottlenecked by what information we can scrape out of `clap` conveniently (and possibly about changing some APIs to make that easier)